.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -111,22 +118,82 @@ func New(c Config) (*PullRequestEnvironment, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// Check if the pull request has changes to the `docs/` directory as that will effect who
+	// the required reviewers are.
+	docChanges, codeChanges, err := hasChanges(c.Context, pr, c.Client)
+	if err != nil {
+		// Log the error, don't fail the whole run because it couldn't detect if the pull request has docs
+		log.Errorf("error while detecting pull request for docs changes: %v, skipping assigning docs reviewers", err)
+	}
+
 	return &PullRequestEnvironment{
 		Client:           c.Client,
 		reviewers:        c.Reviewers,
 		defaultReviewers: c.Reviewers[ci.AnyAuthor],
 		Metadata:         pr,
+		HasDocsChanges:   docChanges,
+		HasCodeChanges:   codeChanges,
 	}, nil
 }
 
+// hasChanges determines if the pull request has changes to `docs/`
+func hasChanges(ctx context.Context, pr *Metadata, clt *github.Client) (hasDocsChanges bool, hasCodeChanges bool, err error) {
+	files, err := getPullRequestFiles(ctx, pr, clt)
+	if err != nil {
+		return false, true, trace.Wrap(err)
+	}
+	for _, file := range files {
+		if file.Filename == nil {
+			return false, true, trace.BadParameter("pull request file name is nil")
+		}
+		if strings.HasPrefix(*file.Filename, ci.DocsPrefix) {
+			hasDocsChanges = true
```

Comment:
I think you might also want to detect any `.md` or `.mdx` suffix changes as docs changes.

Reply:
Or also look for changes in `rfd/`.